### PR TITLE
Devl/new functionality

### PR DESCRIPTION
**Pull request by kelvin-onlab**

### CODE REVIEW

File: TestON/drivers/common/cli/emulator/mininetclidriver.py [hunk 1]
```diff
@@ -95,7 +96,7 @@ def connect( self, **connectargs ):
             main.cleanup()
             main.exit()
 
-    def startNet( self, topoFile='', args='', timeout=120 ):
+    def startNet( self, topoFile='', args='', mnCmd='', timeout=120 ):
         """
         Starts Mininet accepts a topology(.py) file and/or an optional
         argument ,to start the mininet, as a parameter.
```
Comment (jhall11) [inline TestON/drivers/common/cli/emulator/mininetclidriver.py]: Can you modify the doc string to match these changes?

Comment (kelvin-onlab) [inline TestON/drivers/common/cli/emulator/mininetclidriver.py]: Ooops I forgot, yup!

File: TestON/drivers/common/cli/onosclidriver.py [hunk 2]
```diff
@@ -1758,11 +1758,19 @@ def checkIntentState( self, intentsId, expectedState = 'INSTALLED' ):
             returnValue = main.TRUE
             for intents in intentsDict:
                 if intents.get( 'state' ) != expectedState:
-                    main.log.info( self.name + " : " + intents.get( 'id' ) +
-                                   " actual state = " + intents.get( 'state' )
-                                   + " does not equal expected state = "
-                                   + expectedState )
-                    returnValue = main.FALSE
+                    if intents.get( 'state' ) == "INSTALLING":
```
Comment (jhall11) [inline TestON/drivers/common/cli/onosclidriver.py]: I'm not sure this is what we actually want this function to do. 

As it is, we give it the expected state, and return true if the intent is in the expected state or INSTALLING state. This means if we expect the state to be WITHDRAWN the function returns true if the state is WITHDRAWN or INSTALLING. These two states are on opposite ends of the state machine.

What we may want to do is pass in a list of states to expect.

Also, in your test cases, you should be checking the the intent does eventually get into the INSTALLED state and not just the *ING states.

Comment (jhall11) [inline TestON/drivers/common/cli/onosclidriver.py]: We also should update the doc string here

Comment (kelvin-onlab) [inline TestON/drivers/common/cli/onosclidriver.py]: True, I just realize that if the intent is stuck in INSTALLING state this function will still return TRUE and not catch a bug. Thanks! i'll modify it right away as well as add modify the docs.

**Comment by kelvin-onlab** [comment]
Added the assignSwController change.

File: TestON/drivers/common/cli/emulator/mininetclidriver.py [hunk 3]
```diff
@@ -1092,49 +1107,130 @@ def getSwController( self, sw ):
...
-        args = utilities.parse_args( [ "COUNT" ], **kwargs )
-        count = args[ "COUNT" ] if args != {} else 1
-
-        argstring = "SW"
-        for j in range( count ):
-            argstring = argstring + ",IP" + \
-                str( j + 1 ) + ",PORT" + str( j + 1 )
-        args = utilities.parse_args( argstring.split( "," ), **kwargs )
+        Description:
+            Assign switches to the controllers ( for ovs use only )
```
Comment (jhall11) [inline TestON/drivers/common/cli/emulator/mininetclidriver.py]: We have many functions in here that are only for ovs. if/when we start using other software switches, we may want to look into how we can subclass this driver and/or making these functions switch agnostic

File: TestON/drivers/common/cli/emulator/mininetclidriver.py [hunk 4]
```diff
@@ -1092,49 +1107,130 @@ def getSwController( self, sw ):
...
+                                    "assigned and a list of ports has" +
+                                    " been passed" )
+                    return main.FALSE
+                else:
+                    main.log.error( self.name + ": Invalid controller port " +
+                                    "number. Please specify correct " +
+                                    "controller port" )
+                    return main.FALSE
+
+            elif isinstance( ip, types.ListType ):
```
Comment (jhall11) [inline TestON/drivers/common/cli/emulator/mininetclidriver.py]: else clause for this if/else statement

File: TestON/drivers/common/cli/emulator/mininetclidriver.py (hunk 3 above)
Comment (kelvin-onlab) [inline TestON/drivers/common/cli/emulator/mininetclidriver.py]: I agree, almost every function here is for ovs only. We should include other type of switches in our plan in the following week. Curious what other types are there that we can virtually test?

**Comment by jhall11** [comment]
Have you tested these changes?

**Comment by kelvin-onlab** [comment]
Yup~

**Comment by kelvin-onlab** [comment]
Just need to modify the existing function a little bit and its good to go.

**Comment by kelvin-onlab** [comment]
*funtion call

**Comment by jhall11** [comment]
ok, I'd like to merge this before I merge the functions I'm working on so I can make sure all the tests are using the modified versions of the getHosts, getLinks, and getSwitches functions I'm working on and no one is stuck with a broken function call

**Comment by kelvin-onlab** [comment]
Can we merge this to master?

**Pull request merged**